test/retained_topic_map.hpp

```diff
@@ -247,11 +247,34 @@ class retained_topic_map {
     void increase_topics(std::vector<direct_const_iterator> const &path) {
         auto& direct_index = map.template get<direct_index_tag>();
 
-        for(auto i : path) {
+        for(auto& i : path) {
             direct_index.modify(i, [](path_entry &entry){ ++entry.count; });
         }
     }
 
+    // Exceptions used
+    static void throw_max_stored_topics() { throw std::overflow_error("Retained map maximum number of topics reached"); }
+    static void throw_no_wildcards_allowed() { throw std::runtime_error("Retained map no wildcards allowed in retained topic name"); }
+    static void throw_internal_error() { throw std::runtime_error("Retained map internal error"); }
```

Comment:
What does internal_error mean?
If it means logic bug and it shouldn't happen by the invalid input from the users, then it should be BOOST_ASSERT.

Reply:
When reference count drops below zero, which should never happen. So i guess this can be assert.

So it should be assert I guess?

Reply:
Yes. Please update the code.